Added support to follow to next page

lookatitude opened the pull request:
Added support for following links to the next page

lookatitude committed:
Added support to follow to next page

djnnvx commented:
This is very cool, and would be very needed for a tool I'm working on. I would be interested in helping out with this PR if there is changes to make or something, if it can make it go faster.

pjebs commented:
Currently sick with covid. Will look at it after recovery.

lookatitude commented:
just to help clarify the changes ad added code (I didn't want to change a lot.
1 - Added the option to followLinks (if there is a next link at the end of the page we grab the next start value generate the next url and add it to a queue. Added this in a new OnHtml since it has a different selector.
2 - add the first request to the queue and run the queue.
3 - I've renamed the url function to buildUrl as I'm using the url function from net/url package to parse the url params of the next link.

The queue going faster or not will depend on the rate limiter you set. It is a bit slow as you never know if there is a next page or not until you actually scrape the previous page. Well you can but it would make it way more complex to scrape and determain the number of pages, the next link always has the same id so it is easier to grab that one and follow it.

If someone can make it better I welcome the help, this is my first PR in a public go project.

@pjebs Get better soon.
